routers/repo/view.go

```diff
@@ -54,18 +54,38 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
-	var readmeFile *git.Blob
+	var readmes []struct {
+		*git.Blob
+		tp string
+	}
 	for _, entry := range entries {
-		if entry.IsDir() || !markup.IsReadmeFile(entry.Name()) {
+		if entry.IsDir() {
+			continue
+		}
+
+		tp, ok := markup.ReadmeFileType(entry.Name())
+		if !ok {
 			continue
 		}
 
-		// TODO: collect all possible README files and show with priority.
-		readmeFile = entry.Blob()
-		break
+		readmes = append(readmes, struct {
+			*git.Blob
+			tp string
+		}{
+			entry.Blob(),
+			tp,
+		})
 	}
 
-	if readmeFile != nil {
+	if len(readmes) > 0 {
+		var readmeFile *git.Blob
+		for _, readme := range readmes {
```

Comment:
@sapk you are right. The `type` will check which parser could read the README file according ext name. But we also need support `README`, which should be rendered as a TEXT, so the `tp` may be empty. But `README` or `README.diskdefine` is lower priority than `README.md` or `README.org`(this will be implemented at #1527).